### ISSUE TITLE
No clear what is "repository" of algs4

### ISSUE THREAD

**Issue by powazny** [kevin-wayne/algs4#18]
Hello,

What is the Maven dependency and what is the repository we should you to build application using Maven?

**Comment by Momus** [comment]
I am trying to work work through the Coursera class. In my pom.xml file I have
```xml
<dependency>
      <groupId>com.googlecode.princeton-java-algorithms</groupId>
      <artifactId>algorithms</artifactId>
<version>4.0.1</version>
<dependency>   
```
in my code, I have a simple import statement:
```java
import edu.princeton.cs.algs4.StdOut;
```

this does not work with maven! 
```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.1:compile (default-compile) on project algorithms: Compilation failure: Compilation failure:
[ERROR] $HOME/algorithms/src/main/java/org/momus/algorithms/HelloWorld.java:[8,30] package edu.princeton.cs.algs4 does not exist
[ERROR] $HOME/algorithms/src/main/java/org/momus/algorithms/HelloWorld.java:[13,9] cannot find symbol
[ERROR] symbol:   variable StdOut
[ERROR] location: class org.momus.algorithms.HelloWorld
```
Running ` mvn dependency:resolve` shows that it is installed:
````[INFO] Scanning for projects...
[INFO]                                                                         
[INFO] ------------------------------------------------------------------------
[INFO] Building algorithms 0.0.1
[INFO] ------------------------------------------------------------------------
[INFO] 
[INFO] --- maven-dependency-plugin:2.8:resolve (default-cli) @ algorithms ---
[INFO] 
[INFO] The following files have been resolved:
[INFO]    junit:junit:jar:3.8.1:test
[INFO]    java3d:j3d-core:jar:1.3.1:compile
[INFO]    com.googlecode.princeton-java-algorithms:algorithms:jar:4.0.1:compile
[INFO]    java3d:vecmath:jar:1.3.1:compile
[INFO]    java3d:j3d-core-utils:jar:1.3.1:compile
[INFO]    gov.nist.math:jama:jar:1.0.3:compile
[INFO]    com.googlecode.princeton-java-introduction:stdlib:jar:1.0.1:compile
[INFO] 
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 1.066 s
[INFO] Finished at: 2017-01-01T17:03:12-07:00
[INFO] Final Memory: 13M/300M
[INFO] --------------------------------
````
So tried importing it with `import com.googlecode.princeton-java-algorithms.algorithms.StdOut;` and various permutations of this line.
That fails with
```
[ERROR] $HOME/algorithms/src/main/java/org/momus/algorithms/HelloWorld.java:[8,32] ';' expected
[ERROR] -> [Help 1]
```

What do I have to do to use algs4 with Maven?

**Comment by kevin-wayne** [comment]
The Maven and Gradle info was created by Peter Korgan. I don’t use Maven, so can’t provide much more information. 

https://github.com/kevin-wayne/algs4/blob/master/README-MAVEN.txt
https://github.com/kevin-wayne/algs4/blob/master/pom.xml

There is also a bintray link

https://bintray.com/algs4/maven/algs4

If the instructions are unclear, feel free to suggest improvements.



> On Jan 1, 2017, at 7:13 PM, Dmitri Brengauz <notifications@github.com> wrote:
> 
> I am trying to work work through the Coursera class. In my pom.xml file I have
> 
> <dependency
> >
>       <
> groupId>com.googlecode.princeton-java-algorithms</groupId
> >
>       <
> artifactId>algorithms</artifactId
> >
> <
> version>4.0.1</version
> >
> <
> dependency>   
> in my code, I have a simple import statement:
> 
> import edu.princeton.cs.algs4.StdOut;
> this does not work with maven!
> 
> [ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.1:compile (default-compile) on project algorithms: Compilation failure: Compilation failure:
> [ERROR] $HOME/algorithms/src/main/java/org/momus/algorithms/HelloWorld.java:[8,30] package edu.princeton.cs.algs4 does not exist
> [ERROR] $HOME/algorithms/src/main/java/org/momus/algorithms/HelloWorld.java:[13,9] cannot find symbol
> [ERROR] symbol:   variable StdOut
> [ERROR] location: class org.momus.algorithms.HelloWorld
> 
> Running mvn dependency:resolve shows that it is installed:
> 
> [INFO]                                                                         
> [INFO] ------------------------------------------------------------------------
> [INFO] Building algorithms 0.0.1
> [INFO] ------------------------------------------------------------------------
> [INFO] 
> [INFO] --- maven-dependency-plugin:2.8:resolve (default-cli) @ algorithms ---
> [INFO] 
> [INFO] The following files have been resolved:
> [INFO]    junit:junit:jar:3.8.1:test
> [INFO]    java3d:j3d-core:jar:1.3.1:compile
> [INFO]    com.googlecode.princeton-java-algorithms:algorithms:jar:4.0.1:compile
> [INFO]    java3d:vecmath:jar:1.3.1:compile
> [INFO]    java3d:j3d-core-utils:jar:1.3.1:compile
> [INFO]    gov.nist.math:jama:jar:1.0.3:compile
> [INFO]    com.googlecode.princeton-java-introduction:stdlib:jar:1.0.1:compile
> [INFO] 
> [INFO] ------------------------------------------------------------------------
> [INFO] BUILD SUCCESS
> [INFO] ------------------------------------------------------------------------
> [INFO] Total time: 1.066 s
> [INFO] Finished at: 2017-01-01T17:03:12-07:00
> [INFO] Final Memory: 13M/300M
> [INFO] --------------------------------
> 
> So tried importing it with import com.googlecode.princeton-java-algorithms.algorithms.StdOut; and various permutations of this line.
> That fails with
> 
> [ERROR] $HOME/algorithms/src/main/java/org/momus/algorithms/HelloWorld.java:[8,32] ';' expected
> [ERROR] -> [Help 1]
> 
> What do I have to do to use algs4 with Maven?
> 
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub, or mute the thread.
> 

Kevin Wayne
Phillip Y. Goldman '86 Senior Lecturer
Department of Computer Science
Princeton University

**Comment by ny83427** [comment]
@Momus I don't think it needs that many steps, after clone the repository you just need to execute `mvn clean install` and all will be settle down.

```shell
git clone https://github.com/kevin-wayne/algs4 && mvn clean install
```

Then add below to your `pom.xml`:

```xml
        <dependency>
            <groupId>edu.princeton.cs</groupId>
            <artifactId>algs4</artifactId>
            <version>1.0.0.0</version>
        </dependency>
```

**Comment by yudi-azvd** [comment]
The solution above really helped me 👌
